server/models/CustomMetadataProvider.js

```diff
@@ -0,0 +1,58 @@
+const { DataTypes, Model, Sequelize } = require('sequelize')
+
+class CustomMetadataProvider extends Model {
+  constructor(values, options) {
+    super(values, options)
+
+    /** @type {UUIDV4} */
+    this.id
+    /** @type {string} */
+    this.name
+    /** @type {string} */
+    this.url
+    /** @type {string} */
+    this.apiKey
+  }
+
+  getSlug() {
+    return `custom-${this.id}`
+  }
+
+  toUserJson() {
+    return {
+      name: this.name,
+      id: this.id,
+      slug: this.getSlug()
+    }
+  }
+
+  static findByPk(id) {
+    return this.findOne({
+      where: {
+        id,
+      }
+    })
+  }
```

Comment:
```suggestion
```

This function is built-in to Sequelize. https://sequelize.org/docs/v6/core-concepts/model-querying-finders/#findbypk